src/System.Memory/tests/Memory/CustomMemoryForTest.cs

```diff
@@ -38,13 +39,14 @@ public override Span<T> Span
             }
         }
 
-        public override MemoryHandle Pin()
+        public override MemoryHandle Pin(int offset = 0)
         {
             unsafe
             {
                 Retain();
+                if (offset < 0 || (_array.Length > 0 && offset >= _array.Length)) throw new ArgumentOutOfRangeException(nameof(offset));
```

Comment:
I think this should rather be: `if (offset < 0 || offset > _array.Length)`

Reply:
Is it OK to get a pointer past the _array.Length for non-empty arrays?

Reply:
It is ok to get a pointer that points right past the array, but not beyond it. It matches how offsets or index checks are done everywhere else. For example, the condition in Span.Slice is `if ((uint)index > (uint)_count) ... error ...`.